ReportingServicesTools/Functions/CatalogItems/Rest/Write-RsRestCatalogItem.ps1

```diff
@@ -226,37 +239,122 @@ function Write-RsRestCatalogItem
+                        $isLargePowerBIReport = $false
+                    }
+                }
+
+                if ($isLargePowerBIReport -ne $true)
+                {
+                    $payload = @{
+                        "@odata.type" = "#Model.$itemType";
+                        "Content" = [System.Convert]::ToBase64String($fileBytes);
+                        "ContentType"="";
+                        "Name" = $itemName;
+                        "Description" = $Description
+                        "Path" = $itemPath;
+                    }
                 }
             }
 
             try
             {
-                Write-Verbose "Uploading $EntirePath to $RsFolder..."
+                if ($isLargePowerBIReport -ne $true)
+                {
+                    Write-Verbose "Uploading $EntirePath to $RsFolder..."
 
-                $payloadJson = ConvertTo-Json $payload
+                    $payloadJson = ConvertTo-Json $payload
 
-                if ($Credential -ne $null)
-                {
-                    Invoke-WebRequest -Uri $catalogItemsUri -Method Post -WebSession $WebSession -Body ([System.Text.Encoding]::UTF8.GetBytes($payloadJson)) -ContentType "application/json" -Credential $Credential -Verbose:$false | Out-Null
+                    if ($Credential -ne $null)
+                    {
+                        Invoke-WebRequest -Uri $catalogItemsUri -Method Post -WebSession $WebSession -Body ([System.Text.Encoding]::UTF8.GetBytes($payloadJson)) -ContentType "application/json" -Credential $Credential -Verbose:$false | Out-Null
+                    }
+                    else
+                    {
+                        Invoke-WebRequest -Uri $catalogItemsUri -Method Post -WebSession $WebSession -Body ([System.Text.Encoding]::UTF8.GetBytes($payloadJson)) -ContentType "application/json" -UseDefaultCredentials -Verbose:$false | Out-Null
+                    }
                 }
-                else
+
+                if ($itemType -eq "PowerBIReport" -and $isLargePowerBIReport -eq $true)
                 {
-                    Invoke-WebRequest -Uri $catalogItemsUri -Method Post -WebSession $WebSession -Body ([System.Text.Encoding]::UTF8.GetBytes($payloadJson)) -ContentType "application/json" -UseDefaultCredentials -Verbose:$false | Out-Null
+                    Write-Verbose "Uploading $EntirePath to $RsFolder via endpoint for large files..."
+                    
+                    $headers = @{
+                        "accept"="application/json, text/plain, */*"
+                    }
+                    $uri = [String]::Format($powerBIReportsByPathApi, $itemPath)
+
+                    if ($Credential -ne $null)
+                    {
+                        Invoke-RestMethod -Uri $uri -Method Post -WebSession $WebSession -Body $bodyLines -ContentType "multipart/form-data; boundary=$boundary" -Headers $headers -Credential $Credential -Verbose:$false | Out-Null
```

Comment:
this is almost duplicate of the lines 324, instead please extract the things that change and make only one call

for example contentType should be defined and changed to multipart above where you already have the condition for pbi report (line 243)
same for body and headers